### PR TITLE
Add read_entire() and write_entire()

### PR DESCRIPTION
**Pull request by steven-johnson**
Simple utility functions to slurp / spew the entire contents of a file; not much re-use in this PR but I'll need them elsewhere in the future, so this is intended to reduce future PR noise.

### CODE REVIEW

File: src/Util.h [hunk 1]
```diff
@@ -265,6 +265,19 @@ void dir_rmdir(const std::string &name);
 /** Wrapper for stat(). Asserts upon error. */
 FileStat file_stat(const std::string &name);
 
+/** Read the entire contents of a file into a vector<char>. The file
+ * is read in binary mode. Errors trigger an assertion failure. */
+std::vector<char> read_entire(const std::string &pathname);
```
Comment (dsharletg) [inline src/Util.h]: I think these should be named `read_entire_file` and `write_entire_file`, to make the code using it easier to read, and to be consistent with the rest of the file helper functions here.

File: src/Util.cpp [hunk 2]
```diff
@@ -411,6 +412,29 @@ std::string dir_make_temp() {
...
+    f.close();
+    return result;
+}
+
+void write_entire(const std::string &pathname, const void *source, size_t source_len) {
+    std::ofstream f(pathname, std::ios::out|std::ios::binary);
+
+    f.write(reinterpret_cast<const char*>(source), source_len);
+    f.flush();
+    internal_assert(f.good()) << "Unable to write file: " << pathname;
```
Comment (dsharletg) [inline src/Util.cpp]: I think the error checking should be in the caller, so the assert is more meaningful. Same for the read assert, though maybe it means read_entire needs to return a bool and take a reference to a vector as an argument?

File: src/Util.h (hunk 1 above)
Comment (steven-johnson) [reply src/Util.h]: Agreed.

File: src/Util.cpp (hunk 2 above)
Comment (steven-johnson) [reply src/Util.cpp]: Not sure about this; I like the idea of a "can't forget to check the error" function for these purposes.

Comment (steven-johnson) [reply src/Util.cpp]: Also, the other file utilities in this file tend to have the same assert-fail behavior, rather than returning an error code.

**Comment by steven-johnson** [comment]
Going to land this as-is.

**Pull request merged**